### ISSUE TITLE
ArduinoJson for amazon AWS IOT "state:" node

### ISSUE THREAD

**Issue by TLS1000** [bblanchon/ArduinoJson#581]
Hi, I'm using ESP32 with arduino core, and hornbill AWS IOT library.

I'm trying to send an array of floats to AWS but shadow update is refused because of missing "state" node.

The JSON string ArduinoJson generates: 
{"cellVoltages":[3305,2685,3420,3586,3264,3730,3614,3426,2700,3310,3183,3785,2836,2959,3265,3426]}

However AWS IOT refuses this format, according to Hornbill AWS IOT lib example (tested and works) the format should be:
const char payloadFormat[] = "{\"state\":{\"desired\":{\"temperature\":%f,\"irms\":%f,\"power\":%f}}}";

So basically I need to have a string:      {"state":ArduinoJsonstring}

What would be the most elegant way to do this? thank you

**Comment by TLS1000** [comment]
update: I got it working but not elegant:

AWS IOT requires shadow updates with  {"state":"reported":ArduinoJsonstring}
You need attributes "reported" for things communicating their state, and "desired" for setpoints coming from the cloud.

the 'not elegant' part is that i used a second char array 'tempbuffer' the same size as the jsonbuffer:
sprintf(tempbuffer,"{\"state\":{\"reported\":%s}}",ArduinoJson);

It would be nice if I could use the ArduinoJson-string directly in AWS IOT publish like this:
if(hornbill.publish(TOPIC_NAME,ArduinoJson) == 0)

**Comment by TLS1000** [comment]
got it working, elegant!

 StaticJsonBuffer<250> jsonBuffer;
  JsonObject& root=jsonBuffer.createObject();
  JsonObject& state=root.createNestedObject("state");
  JsonObject& reported=state.createNestedObject("reported");
  JsonArray& cellVoltages = reported.createNestedArray("cellVoltages");
  for(int i=0;i<16;i++){
    cellVoltages.add(random(2400,3800));
  }
  root.prettyPrintTo(Serial);
  root.printTo(payload,sizeof(payload));
  if(hornbill.publish(TOPIC_NAME,payload) == 0)     //send the ArduinoJsonArray to AWS IOT
